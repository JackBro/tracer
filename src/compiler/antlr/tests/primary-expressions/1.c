void main(){
	int *x, y, a[5];

	(a[2]) = x;

	(y) = 5;

	(a[3]) = (a[2]) + 4;
	
	(y) = (y) + 2;
}

