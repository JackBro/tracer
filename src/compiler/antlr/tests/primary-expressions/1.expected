trans(func_main, i5,
	[p$init_0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p$init_1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[],
	[],
	[],
	not_branch).
trans(func_main, i6,
	[p$init_1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[],
	[call(func___TRACER_INIT,[RV0|_],[SRV0|_],[])],
	[],
	not_branch).
trans(func_main, i1,
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p7],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A$Next$],[ReturnValue,X,Y$Next$],[SA$Next$],[SReturnValue,SX,SY$Next$],
	[arrassign(SA$1$,upd(SA,2,SX)),assign(SY$1$,5),arrassign(SA$Next$,upd(SA$1$,3,plus(ref(SA$1$,2),4))),assign(SY$Next$,plus(SY$1$,2))],
	[],
	[ref(A,2,ARef0),ref(A$1$,3,A$1$Ref0),ref(A$1$,2,A$1$Ref1)],
	not_branch) :-
	A$1$=upd(A,2,X),(Y$1$)=5,A$Next$=upd(A$1$,3,(A$1$Ref1)+4),(Y$Next$)=(Y$1$)+2.
trans(func_main, i2,
	[p7],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],
	[p8],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue$Next$,X,Y],[SA],[SReturnValue$Next$,SX,SY],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
trans(func___TRACER_INIT, i3,
	[p0],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[p1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[],
	[],
	[],
	not_branch).
trans(func___TRACER_INIT, i4,
	[p1],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue],[],[SReturnValue],
	[p2],[H$1$,H$2$],[],[SH$1$,SH$2$],[],[],[ReturnValue$Next$],[],[SReturnValue$Next$],
	[assign(SReturnValue$Next$,_)],
	[],
	[],
	not_branch) :-
	ReturnValue$Next$=_.
%--------------------------------------------------
block(b1, func_main, [_], [_], _, nl, [i1,i2,i5,i6],  [n,n],[],[u],[u,n,u]).
block(b2, func___TRACER_INIT, [_], [_], _, nl, [i3,i4],  [n,n],[],[],[u]).
%--------------------------------------------------
start(func_main, [p$init_0]).
start(func___TRACER_INIT, [p0]).
%--------------------------------------------------
bottom(func_main, [p8]).
bottom(func___TRACER_INIT, [p2]).
%--------------------------------------------------
freshglobalarrays([SH$1$,SH$2$]).
%--------------------------------------------------
freshglobalvars([]).
%--------------------------------------------------
freshlocalarrays(func_main,[SA]).
freshlocalarrays(func___TRACER_INIT,[]).
%--------------------------------------------------
freshlocalvars(func_main,[SReturnValue,SX,SY]).
freshlocalvars(func___TRACER_INIT,[SReturnValue]).
%--------------------------------------------------
freshparams(func_main,[lv(0)]).
freshparams(func___TRACER_INIT,[lv(0)]).
%--------------------------------------------------
init_goal(func_main, [p$init_0], i5, [H$1$,H$2$],[],[SH$1$,SH$2$],[],[A],[ReturnValue,X,Y],[SA],[SReturnValue,SX,SY],[]).
